src/Orleans.Transactions/State/TransactionManager.cs

```diff
@@ -15,8 +15,9 @@ public TransactionManager(TransactionQueue<TState> queue)
             this.queue = queue ?? throw new ArgumentNullException(nameof(queue));
         }
 
-        public Task<TransactionalStatus> PrepareAndCommit(Guid transactionId, AccessCounter accessCount, DateTime timeStamp, List<ParticipantId> writeResources, int totalResources)
+        public async Task<TransactionalStatus> PrepareAndCommit(Guid transactionId, AccessCounter accessCount, DateTime timeStamp, List<ParticipantId> writeResources, int totalResources)
         {
+            await this.queue.Ready();
```

Comment:
no need to wait here. (If grain is undergoing restore, lock validation is guaranteed to fail.)